apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -37,66 +38,96 @@ const WEEKS_IN_YEAR = 52;
 const SECONDS_IN_DAY = 86400;
 const SECONDS_IN_YEAR = 365 * SECONDS_IN_DAY;
 
+type PoolSnapshot = {
+  timestamp: number;
+  liquidity: string;
+  swapVolume: string;
+  pool: {
+    symbol?: string | null;
+    tokens?: { symbol: string; balance: string }[] | null;
+  };
+  swapFees?: string;
+};
+
 async function fetchPoolAveragesForDateRange(
   poolId: string,
   network: string,
   from: number,
   to: number,
 ): Promise<[number, number, string, { symbol: string; balance: string }[]]> {
+  // Determine if the initial date range is less than 2 days
+  const initialRangeInDays = calculateDaysBetween(from, to);
+  const extendedFrom = initialRangeInDays < 2 ? from - SECONDS_IN_DAY : from;
+
+  // Fetch snapshots within the (potentially extended) date range
   const res = await pools.gql(network).poolSnapshotInRange({
     poolId,
-    from,
+    from: extendedFrom,
     to,
   });
 
-  res.poolSnapshots = res.poolSnapshots.sort(
+  if (res.poolSnapshots.length === 0) {
+    console.warn(
+      `No data found for pool ${poolId}(${network}) in range ${from} - ${to}`,
+    );
+    return [0, 0, "", []];
+  }
+
+  const sortedSnapshots: PoolSnapshot[] = res.poolSnapshots.sort(
     (a, b) => a.timestamp - b.timestamp,
   );
 
-  if (res.poolSnapshots.length === 0) {
-    // Following Fabio's recomendation on #BAL-872
-    if (calculateDaysBetween(from, to) != 1) {
-      return [0, 0, "", []];
+  if (
+    sortedSnapshots.length <= 1 &&
+    sortedSnapshots[sortedSnapshots.length - 1].timestamp <= to
+  ) {
+    const currentData = await pools.gql(network).Pool({ poolId });
```

Comment:
This is supposed to be the workaround when no snapshot doesn't has data for a given day, right? If so why are we fetching data from the current day?

Reply:
Was fixed on #329, This PRs breaks again